projects/ion/src/lib/accordion/accordion.component.ts

```diff
@@ -1,27 +1,60 @@
-import { Component, Input, OnInit } from '@angular/core';
-import { TemplateHeader } from '../core/types/accordion';
+import {
+  Component,
+  TemplateRef,
+  Input,
+  Output,
+  EventEmitter,
+  OnInit,
+} from '@angular/core';
+import { AccordionItem, AccordionItemOutput } from '../core/types/accordion';
 
 @Component({
   selector: 'ion-accordion',
-  templateUrl: './accordion.component.html',
-  styleUrls: ['./accordion.component.scss'],
+  templateUrl: 'accordion.component.html',
+  styleUrls: ['accordion.component.scss'],
 })
 export class IonAccordionComponent implements OnInit {
-  @Input() name?: string;
-  @Input() templateHeader?: TemplateHeader;
-  @Input() show? = false;
-
-  iconSize = 24;
+  @Input() accordions: AccordionItem[] = [];
+  @Input() templateBody: TemplateRef<HTMLElement>;
+  @Input() modeAccordion = true;
+  @Input() templateHeader: TemplateRef<HTMLElement>;
+  @Output() activeChange = new EventEmitter<AccordionItemOutput>();
 
   ngOnInit(): void {
-    if (!this.name && !this.templateHeader) {
-      throw new Error(
-        'The name or templateHeader properties were not set correctly'
-      );
+    this.validate();
+  }
+
+  toggle(index: number): void {
+    const state: boolean = this.accordions[index].show;
+
+    if (this.modeAccordion) {
+      this.closeAccordions();
     }
+
+ this.accordions[index] = {
+   ...this.accordions[index],
+   show: !state,
+   key: index
+ }
```

Comment:
Sorry, I not indent this code in my review.

Reply:
No problem! :smile: